File: Task 4/Java/app/src/main/java/dev/ogabek/java/manager/LocaleManager.java
package dev.ogabek.java.manager;

import android.content.Context;
import android.content.res.Configuration;
import android.content.res.Resources;
import android.content.res.TypedArray;
import android.os.Build;
import android.os.LocaleList;

import androidx.annotation.RequiresApi;

import java.util.LinkedHashSet;
import java.util.Locale;

public class LocaleManager {

    Context context;

    public LocaleManager(Context context) {
        this.context = context;
    }

    public static final String LANGUAGE_ENGLISH = "en";
    public static final String LANGUAGE_RUSSIAN = "ru";
    public static final String LANGUAGE_UZBEK = "uz";

    public void updateLanguage(Context context, String language) {
        Locale locale = new Locale(language);
        Locale.setDefault(locale);
        Configuration config = new Configuration();
        config.locale = locale;
        context.getResources().updateConfiguration(config, context.getResources().getDisplayMetrics());

        new PrefsManager(context).saveData("language", language);

    }


    private boolean isAtLeastVersion(int version) {
        return Build.VERSION.SDK_INT >= version;
    }


}

File: Task 4/Kotlin/app/src/main/java/dev/ogabek/kotlin/manager/LocaleManager.kt
package dev.ogabek.kotlin.manager

import android.content.Context
import android.content.res.Configuration
import android.os.Build
import java.util.*

class LocaleManager(var context: Context) {
    fun updateLanguage(context: Context, language: String?) {
        val locale = Locale(language)
        Locale.setDefault(locale)
        val config = Configuration()
        config.locale = locale
        context.resources.updateConfiguration(config, context.resources.displayMetrics)
        PrefsManager(context).saveData("language", language)
    }

    private fun isAtLeastVersion(version: Int): Boolean {
        return Build.VERSION.SDK_INT >= version
    }

    companion object {
        const val LANGUAGE_ENGLISH = "en"
        const val LANGUAGE_RUSSIAN = "ru"
        const val LANGUAGE_UZBEK = "uz"
    }
}